channeldb/reports.go

```diff
@@ -50,6 +50,18 @@ const (
 	// ResolverOutcomeUnknown is a placeholder for unknown resolver
 	// outcomes.
 	ResolverOutcomeUnknown ResolverOutcome = iota
+
```

Comment:
Ah ok, the type is essentially encoded in the outcome.

Reply:
Yeah, couldn't see much case for storing type and outcome separately, since we'd prob just need to map them to a single enum over rpc anyway.

Reply:
In general enums can always be combined, but I am not convinced that that is a good thing to do. You'll need to define all combinations of values in a single enum. It is visible also in the suffixes of this enum. The suffixes also aren't fully consistent (`Claimed`, `Claim`, `Recovered` for example).

Also on the RPC I wouldn't map them to a single enum.

Reply:
I'd be for splitting up enums into type and outcome if every type/outcome combination make sense, but in this case there are some types that can't logically be combined with an outcome (eg anchors don't have success txns, outgoing htlcs don't have the same outcomes as anchors). 

Cleaned up the naming here to be more consistent, but still separate unless it's a big blocker.

Reply:
Even if not every combination makes sense, it can still be better to separate them. I'd think that the majority of combinations is valid. The `Unclaimed` state for example can also make sense for other resolvers where we sometimes abandon resolution even though the final tx (by the remote) hasn't landed.